### ISSUE TITLE
mod_creditCalc

### ISSUE THREAD

**Issue by tankista01** [spoter/spoter-mods#75]
Hi, 

I have some question - may it's not an issue, but I don't know how can I write message in other way :) . 
I stared using  mod_creditCalc and found that for some tanks default coeff is completely different that the true one. I found, that in mod scripts there is a section, where all coeffs are written as comments. And here is my question. How do you get all this values? Are they saved in some game's files? I have tried to calculate them by myself and I analysed over 80k replays to get them, and in most cases there are equal to value in this mod. But in some lines of code there is information, that coeff was changed in some patch and it was fixed. Could you tell me, how do you know if something was changed and what the new value is? I think you are not analysing  thousands of replay like me ;). Is there any other way to get value of coeff value?

**Comment by spoter** [comment]
in may hangar 634 vehicles. im just play on all without damage 1 battle. all coeffs calced and saved once.
if changed in next wot update, need play 1 battle without damage

**Comment by tankista01** [comment]
Wow, I thought you would have a nicer way to calculate values. At the beginning I thought in same way but playing all the vehicles ... it would be terrible. After some time I have realised, that in battle result is info about credits for all players in battle, so now I extract this data from replays and after it I am calculating dominant of values (I treat player damage as done by him after his spot) but at beginning I was playing 1 battle without damage and found, that my coeff sometimes is completely different than your (here is my excel file: https://drive.google.com/file/d/1m3LoiBbDq04a28YL2wUyHGbhk42gE2lv/view  In some cases differents are huge).
But I have one more question. How do you know if coeff was changed in update? The only way to know it is when credits after battle are different than calculated ones.

**Comment by spoter** [comment]
https://wiki.wargaming.net/en/Battle_Mechanics#Tank_Experience_and_Credits
you method good if vehicle done battle without damage. we dont know when you shot on self spot or ally spot.

**Comment by tankista01** [comment]
I know that we don't know it, but I assumed that if player has battle without damage or battle where all shots are made on self spot the coeff should be the same. If there are some shots that are made on ally spot the calculated coeff will be different and this cases coeff value will be more varied and calculating dominant will give as true coeff due to greater repeatability (it's easier to find battle where all shots was made on self spot than find battle where player made some shots on ally spot and coeff is the same for different battle). If you look at my excel, there are many records that my and your coeff is the same - so my method look fine. But there are some different. e.g polish I tier 4TP. When I have played battle without damage with this tank and lose battle I get 701 credits. This results is much closer to my coeff than yours: 1.419.

At the and I want to ask you again. How do you know if coeff was changed for some tanks. Did you play all yours tanks with 0 dmg after every patch? :)

**Comment by spoter** [comment]
no. im play only once, when create data. ~1000 batles without damage, crazy wn8. im never repeat it last year.

**Comment by tankista01** [comment]
Ok. I understand, bot how do you know if WG changed some coeff? I saw some notification in code, that some coeff was fixed due to change in some patch

**Comment by spoter** [comment]
according to the results of the battle, in which no damage was inflicted, automatically makes a change in the coefficients and writes in the game log

**Comment by tankista01** [comment]
Do you play 0 dmg battles on purpose or if it comes out by chance? I mean, do you update the coefficient because you know it has changed, or maybe you know it has changed because you accidentally played a battle with 0 dmg and a new, different coefficient was calculated?

**Comment by spoter** [comment]
every new battle with 0 damage, coeff calculated and check. if coeff changed = saved in res_mods/<version>/system/sw_templates.json
